compiler/rustc_resolve/src/imports.rs

```diff
@@ -693,16 +749,34 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
                 format!("`{path}`")
             })
             .collect::<Vec<_>>();
-        let msg = format!("unresolved import{} {}", pluralize!(paths.len()), paths.join(", "),);
+        let msg = if errors.len() == 1
+            && let Some(message) =
+                errors[0].1.on_unknown_item_attr.as_mut().and_then(|a| a.message.take())
+        {
+            message
+        } else {
+            format!("unresolved import{} {}", pluralize!(paths.len()), paths.join(", "),)
```

Comment:
Like with all other diagnostic attributes, the original messages need to be preserved in notes